### DISCUSSION TITLE
Why is tracking with space-chage in Ocelot much faster than ASTRA?

### DISCUSSION THREAD

**Discussion by bprach** [ocelot-collab/ocelot#118] [General]
I did a 4 nC electron bunch (with momentum of 15 MeV/c) tracking simulation through drifts and QMs in total 16m length. I used ASTRA and Ocelot for comparison, and the results are comparable, even almost identical, as shown in the attached PDF file.  
[TestS2E_CompareASTRAandOcelot.pdf](https://github.com/ocelot-collab/ocelot/files/7588688/TestS2E_CompareASTRAandOcelot.pdf)

I had a look in _Tutorial N3. Space Charge_.  It mentions that ASTRA and Ocelot use the same 3D space-charge algorithm. However, the simulation with ASTRA took ~2.5 hours to finish but Ocelot took only ~5 minutes for the same simulation.

My questions are Why was Ocelot much faster? Is it expected  or I maybe did simulation setup wrongly?

**Comment by sergey-tomin** [comment]
Hi, 
the point is OCELOT uses matrices up to 2nd order while ASTRA use Runge-Kutta integrator for particle tracking.
Cheers,
Sergey.

**Reply by bprach** [reply]
Thanks!
